contracts/peripheral/ChildGaugeVoteRewards.sol

```diff
@@ -0,0 +1,150 @@
+    modifier onlyDistributor() {
+        require(msg.sender == distributor, "!distributor");
+        _;
+    }
+
+    /* -------------------------------------------------------------------
+       Setters 
+    ------------------------------------------------------------------- */
+
+    function setDistributor(address _distributor) external onlyOwner {
+        distributor = _distributor;
+    }
+
+    function setPoolIds(uint256[] memory _poolIds) external {
+        uint256 poolIdsLen = _poolIds.length;
+        for (uint256 i = 0; i < poolIdsLen; i++) {
+            uint256 pid = _poolIds[i];
+            IBooster.PoolInfo memory poolInfo = booster.poolInfo(pid);
+            getPoolId[poolInfo.gauge] = pid;
+        }
+    }
+
+    /* -------------------------------------------------------------------
+       Core 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @notice Process gauge rewards for the given epoch for mainnet
+     * @param _gauge Array of gauges
+     * @param _epoch The epoch
+     */
+    function processGaugeRewards(address[] calldata _gauge, uint256 _epoch) external onlyDistributor {
+        for (uint256 i = 0; i < _gauge.length; i++) {
+            address gauge = _gauge[i];
+
+            // Send pro rate AURA to this stash
+            uint256 amountToSend = _getAmountToSend(_epoch, gauge);
+
+            // Fund the extra reward distro for the next 2 epochs
+            uint256 pid = getPoolId[gauge];
+            stashRewardDistro.fundPool(pid, address(aura), amountToSend, 2);
+        }
+    }
+
+    /* -------------------------------------------------------------------
+       Internal 
+    ------------------------------------------------------------------- */
+
+    function _getAmountToSend(uint256 _epoch, address _gauge) internal returns (uint256) {
+        // Send pro rata AURA to the sidechain
+        uint256 amountToSend = getAmountToSendByEpoch[_epoch][_gauge];
+        require(amountToSend != 0, "amountToSend=0");
+
+        // Prevent amounts from being sent multiple times
+        uint256 amountSent = getAmountSentByEpoch[_epoch][_gauge];
+        require(amountSent == 0, "amountSent!=0");
+        getAmountSentByEpoch[_epoch][_gauge] = amountToSend;
+
+        return amountToSend;
+    }
+
+    function _blockingLzReceive(
+        uint16,
+        bytes memory,
+        uint64,
+        bytes memory _payload
+    ) internal override {
+        // Decode the array of payloads
+        (uint256 epoch, bytes[] memory payloads) = abi.decode(_payload, (uint256, bytes[]));
+        uint256 payloadsLen = payloads.length;
+
+        for (uint256 i = 0; i < payloadsLen; i++) {
+            bytes memory payload = payloads[i];
+
+            // Decode the single payload to get the gauge and amount to send
+            (address gauge, uint256 amountToSend) = abi.decode(payload, (address, uint256));
+            getAmountToSendByEpoch[epoch][gauge] = amountToSend;
```

Comment:
It can't get called more than once for an epoch for a gauge you can see lzSend in GaugeVoteRewards. We should make a test case to ensure this is covered.